src/playground-code-editor.ts

```diff
@@ -384,17 +411,252 @@ export class PlaygroundCodeEditor extends LitElement {
+
+    const hintList = this._completionsAsHints();
+
+    const hints = {
+      from: {line: lineNumber, ch: token.start} as Position,
+      to: {line: lineNumber, ch: token.end} as Position,
+      list: hintList,
+    } as Hints;
+
+    CodeMirror.on(
+      hints,
+      'select',
+      async (hint: Hint | string, element: Element) => {
+        if (!this._isCodeEditorHint(hint)) return;
+        // If the current selection is the same, e.g. the completions were just
+        // updated by user input, instead of moving through compltions, we don't
+        // want to re-render and re-fetch the details.
+        if (this._currentSelectionLabel === hint.text) return;
+
+        this._onSelectedChange?.();
+
+        this._renderHint(element as HTMLElement, hints, hint, hint.details);
+      }
+    );
+
+    return hints;
+  }
+
+  private _isCodeEditorHint(hint: Hint | string): hint is CodeEditorHint {
+    return (
+      typeof hint !== 'string' &&
+      Object.prototype.hasOwnProperty.call(hint, 'details')
+    );
+  }
+
+  private _renderHint(
+    element: HTMLElement | undefined,
+    _data: Hints,
+    hint: CodeEditorHint,
+    detail?: Promise<EditorCompletionDetails>
+  ) {
+    if (!element) return;
+
+    const itemIndex = _data.list.indexOf(hint);
+    const completionData = this.completions?.[itemIndex];
+    const objectName = this._buildHintObjectName(
+      hint.displayText,
+      completionData
+    );
+    // Render the actual completion item first
+    this._renderCompletionItem(objectName, element);
+
+    // And if we have the detail promise passed into this function,
+    // we want to asynchronously update the detail info into our completion
+    // item. We don't want to block the rendering, so we don't use await.
+    if (detail !== undefined) {
+      detail.then((detailResult: EditorCompletionDetails) => {
+        this._renderCompletionItemWithDetails(
+          objectName,
+          detailResult,
+          element
+        );
+        // Set the current onSelectedChange to a callback to re-render
+        // the currently selected element, but without the details. This is
+        // then triggered when moving to another selection, removing the details
+        // text from the previously selected element.
+        this._onSelectedChange = () => this._renderHint(element, _data, hint);
+        this._currentSelectionLabel = hint.text;
+      });
+    }
+  }
+
+  private _renderCompletionItem(
+    objectName: DirectiveResult,
+    target: HTMLElement
+  ) {
+    render(html`<span class="hint-object-name">${objectName}</span>`, target);
+  }
+
+  private _renderCompletionItemWithDetails(
+    objectName: DirectiveResult,
+    details: EditorCompletionDetails,
+    target: HTMLElement
+  ) {
+    render(
+      html`<span class="hint-object-name">${objectName}</span>
+        <span class="hint-object-details">${details.text}</span> `,
+      target
+    );
+  }
+
+  private _buildHintObjectName(
```

Comment:
Can you document this method a bit? Not 100% sure what's going on here without digging deeper. How is the result being used, how is the `<mark>` being used, why is there an issue with padding, etc.